generators/app/index.js

```diff
@@ -11,18 +11,30 @@ function sortByKeys(a, b) {
 }
 
 module.exports = yeoman.generators.Base.extend({
+  constructor: function () {
+    yeoman.generators.Base.apply(this, arguments);
+
+    // destination path
+    this.option('path', {
+      type: String,
+      required: false,
+      defaults: '',
+      desc: 'Use an other destination path.'
+    });
+  },
+
   writing: {
     app: function () {
       var optional =  this.options.config || {};
-      var existing = this.fs.exists(this.destinationPath('.travis.yml'))
-            ? yaml.parse(this.fs.read(this.destinationPath('.travis.yml')))
+      var existing = this.fs.exists(this.destinationPath(this.options.path + '.travis.yml'))
```

Comment:
The node way of concatenating path is `path.join(a, b, c?)`.

Using `+`, you're relying on the user to provide a `/` which will not always be the case.